provider/semgrep/client.ts

```diff
@@ -0,0 +1,55 @@
+import type { Finding, Findings } from './api.js'
+import type { Settings } from './index.js'
+
+export default class API {
+    private signal: any
+    private url: string
+    private repo: string
+    private token: string
+    private deployment: string
+
+    constructor(settings: Settings) {
+        this.signal = null
+        this.token = settings.token
+        this.repo = settings.repo || ''
+        this.deployment = settings.deployment
+        this.url = 'https://semgrep.dev/api/v1'
+    }
+
+    private abort(): any {
+        if (this.signal === null) {
+            const abc = new AbortController()
+            setTimeout(() => abc.abort(), 5000)
+            this.signal = abc.signal
```

Comment:
Would reusing the abortcontroller here mean that it would be cancelled for followup requests? (not a huge expert with abortcontrollers)

Reply:
honestly, I'm not sure either. will do some reading up re this.

Reply:
looks like reusing abortcontrollers is a bad idea -- reverting.